src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java

```diff
@@ -13,6 +13,15 @@
  */
 @Value
 class CategoryOrChatUserListItem {
```

Comment:
While refactoring this class, why not offer either individual constructors only taking a category or a user or offering static builder function `of(ChatUserCategory category)` and  `of(ChatChannelUser user)`